sdk/identity/identity/samples/AzureIdentityExamples.md

```diff
@@ -908,6 +915,215 @@ class RotatingCertificateCredential implements TokenCredential {
+import fs from "fs";
+import { promisify } from "util";
+import path from "path";
+
+useIdentityExtension(cachePersistenceExtension);
+
+export async function main(): Promise<void> {
+  const AUTH_RECORD_PATH = "./tokencache.bin";
+  const readFileAsync = promisify(fs.readFile);
+  const fileContent = await readFileAsync(path.join(process.cwd(), AUTH_RECORD_PATH), {
+    encoding: "utf-8"
+  });
+  const authRecord: AuthenticationRecord = deserializeAuthenticationRecord(fileContent);
+
+  const credential = new InteractiveBrowserCredential({
+    tokenCachePersistenceOptions: {
+      enabled: true
+    },
+    authenticationRecord: authRecord
+  });
+}
+
+main().catch((err) => {
+  console.log("error code: ", err.code);
+  console.log("error message: ", err.message);
+  console.log("error stack: ", err.stack);
+});
+```
+
+The credential created in this example will silently authenticate given that a valid token for corresponding to the `AuthenticationRecord` still exists in the persisted token data. There are some cases where interaction will still be required such as on token expiry, or when additional authentication is required for a particular resource.
+
+### Persisting credentials by configuring TokenCachePersistenceOptions
+
+Many credential implementations in the `@azure/identity` library have an underlying token cache which persists sensitive authentication data such as account information, access tokens, and refresh tokens. By default this data exists in an in memory cache which is specific to the credential instance. However, there are scenarios where an application needs persist it across executions in order to share the token cache across credentials. To accomplish this the `@azure/identity` provides the `tokenCachePersistenceOptions`.
+
+> IMPORTANT! The token cache contains sensitive data and MUST be protected to prevent compromising accounts. All application decisions regarding the persistence of the token cache must consider that a breach of its content will fully compromise all the accounts it contains.
+
+#### Using the default token cache
+
+The simplest way to persist the token data for a credential is to to use the default `tokenCachePersistenceOptions`. This will persist and read token data from a shared persisted token cache protected to the current account.
+
+```ts
+import { useIdentityExtension, InteractiveBrowserCredential } from "@azure/identity";
+import { cachePersistenceExtension } from "@azure/identity-cache-persistence";
+
+useIdentityExtension(cachePersistenceExtension);
+
+const credential = new InteractiveBrowserCredential({
+  tokenCachePersistenceOptions: {
+    enabled: true
+  }
+});
+```
+
+#### Using a named token cache
+
+Some applications may prefer to isolate the token cache they use rather than using the shared instance. To accomplish this they can specify the `tokenCachePersistenceOptions` when creating the credential and provide a `name` for the persisted cache instance.
```

Comment:
Nit: "Shared instance" might be a little misleading (between what/whom?). Maybe "default token cache" is better.

Reply:
removed